src/components/MetadataHookForm/useMetadataFormControls.test.ts

```diff
@@ -0,0 +1,330 @@
+import { ChangeEvent } from "@dashboard/hooks/useForm";
+import { act, renderHook } from "@testing-library/react-hooks";
+import { useForm } from "react-hook-form";
+import { IntlShape, MessageDescriptor } from "react-intl";
+
+import { EventDataAction, EventDataField, MetadataFormData } from "../Metadata/types";
+import { useMetadataFormControls } from "./useMetadataFormControls";
+
+jest.mock("react-intl", () => ({
+  useIntl: (): Pick<IntlShape, "formatMessage"> => ({
+    formatMessage: ({ defaultMessage }: MessageDescriptor) => defaultMessage || "",
+  }),
+}));
+describe("useMetadataFormControls", () => {
+  it("returns form values from hook form to be used in form", () => {
+    const formMethods = renderHook(() =>
+      useForm<MetadataFormData>({
+        values: {
+          metadata: [{ key: "key", value: "value" }],
+          privateMetadata: [{ key: "private-key", value: "private-value" }],
+        },
+      }),
+    );
```

Comment:
question: Is it necessary to render an external library in the test? Why not mock them?

Reply:
I think it's worth leaving it for *this specific test* - we can at least test if we're passing data correctly do react-hook-form and if the metadata values are in fact rendering or not